src/abi_arm.cpp

```diff
@@ -0,0 +1,102 @@
+// This file is a part of Julia. License is MIT: http://julialang.org/license
+//
+// The ABI implementation used for 32 bit ARM targets.
+
+#ifndef __ARM_EABI__
+#  error "the Julia ARM ABI implementation only supports EABI"
+#endif
+
+#ifndef __ARM_PCS_VFP
+#  error "the Julia ARM ABI implementation requires VFP support"
+#endif
+
+typedef bool AbiState;
+AbiState default_abi_state = 0;
+
+inline bool isHomogeneousAggregate(jl_value_t *ty, const jl_value_t *&base,
+                                   size_t &members) {
```

Comment:
this implementation doesn't match the ARM definition of a "Homogeneous Aggregates", due to the restriction that "The test for homogeneity is applied after data layout is completed and without regard to access control or other source language restrictions" (whereas this is being done with regards to the source language types).

e.g. the example given in the docs says that a vector of 8 x i8 and a vector of 4 x i16 should be considered homogeneous since they are both the same fundamental ARM type (containerized vector) and size (64 bytes)